raiden_contracts/tests/test_unit_internals.py

```diff
@@ -0,0 +1,246 @@
+
+
+def test_min_uses_usigned(token_network_test):
+    """ Min cannot be called with negative values. """
+    INVALID_VALUES = [-UINT256_MAX, -1]
+    VALID_VALUES = [UINT256_MIN, UINT256_MAX, UINT256_MAX]
+
+    all_invalid = chain(
+        product(VALID_VALUES, INVALID_VALUES),
+        product(INVALID_VALUES, VALID_VALUES),
+    )
+
+    for a, b in all_invalid:
+        with pytest.raises(ValidationError):
+            token_network_test.functions.minPublic(a, b).call()
+
+
+def test_max_uses_usigned(token_network_test):
+
+    INVALID_VALUES = [-UINT256_MAX, -1]
+    VALID_VALUES = [UINT256_MIN, UINT256_MAX, UINT256_MAX]
+
+    all_invalid = chain(
+        product(VALID_VALUES, INVALID_VALUES),
+        product(INVALID_VALUES, VALID_VALUES),
+    )
+    for a, b in all_invalid:
+        with pytest.raises(ValidationError):
+            token_network_test.functions.maxPublic(a, b).call()
+
+
+def test_min(token_network_test):
+
+    VALUES = [UINT256_MIN, 1, UINT256_MAX, UINT256_MAX]
+    for a, b in product(VALUES, VALUES):
+        assert token_network_test.functions.minPublic(a, b).call() == min(a, b)
+
+
+def test_max(token_network_test):
+
+    VALUES = [UINT256_MIN, 1, UINT256_MAX, UINT256_MAX]
+    for a, b in product(VALUES, VALUES):
+        assert token_network_test.functions.maxPublic(a, b).call() == max(a, b)
+
+
+def test_recoverAddressFromWithdrawMessage(
+        token_network_test,
+        create_withdraw_signatures,
+        create_channel_and_deposit,
+        get_accounts
+):
+
+    (A, B) = get_accounts(2)
+    deposit_A = 5
+    deposit_B = 7
+    withdraw_A = 3
+    channel_identifier = create_channel_and_deposit(A, B, deposit_A, deposit_B)
+    (signature_A, signature_B) = create_withdraw_signatures(
+        [A, B],
+        channel_identifier,
+        A,
+        withdraw_A,
+        token_network_test.address
+    )
+    token_network_test.functions.verifyWithdrawSignaturesPublic(
+        channel_identifier,
+        A,
+        B,
+        3,
+        signature_A,
+        signature_B
+    ).call()
+
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            signature_B,
+            signature_A
+        ).call()
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            signature_A,
+            fake_bytes(64)
+        ).call()
+    with pytest.raises(TransactionFailed):
+        token_network_test.functions.verifyWithdrawSignaturesPublic(
+            channel_identifier,
+            A,
+            B,
+            3,
+            fake_bytes(64),
+            signature_B
+        ).call()
+
+    recovered_address_A = token_network_test.functions.recoverAddressFromWithdrawMessagePublic(
```

Comment:
This should be a separate test. You are testing another function, also partially tested in https://github.com/raiden-network/raiden-contracts/blob/5468fcdc249644838d21ef1e90849b6aa1273b20/raiden_contracts/tests/test_channel_withdraw.py#L17

Reply:
Agreed.